From e5f8360a5fa0d52548c205156a23dbf2a66969f7 Mon Sep 17 00:00:00 2001
From: Liu Ying <Ying.Liu@freescale.com>
Date: Tue, 25 Jan 2011 11:52:57 +0800
Subject: [PATCH] ENGR00138468-1 MX5 video:Disable DP/DC/DI/IDMAC before go to kernel

This patch fixes the kernel bootup random hang issue by disabling
DP/DC/DI/IDMAC before we go to kernel. This is a workaround.

Signed-off-by: Liu Ying <Ying.Liu@freescale.com>
---
 cpu/arm_cortexa8/cpu.c |    8 ++++++++
 1 files changed, 8 insertions(+), 0 deletions(-)

diff --git a/cpu/arm_cortexa8/cpu.c b/cpu/arm_cortexa8/cpu.c
index e4bd7cf..70590ee 100644
--- a/cpu/arm_cortexa8/cpu.c
+++ b/cpu/arm_cortexa8/cpu.c
@@ -33,6 +33,9 @@
 
 #include <common.h>
 #include <command.h>
+#ifdef CONFIG_VIDEO_MX5
+#include <ipu.h>
+#endif
 #include <asm/system.h>
 #include <asm/cache.h>
 #include <asm/cache-cp15.h>
@@ -131,6 +134,11 @@ int cleanup_before_linux(void)
 {
 	unsigned int i;
 
+#ifdef CONFIG_VIDEO_MX5
+	ipu_disable_channel(MEM_BG_SYNC);
+	ipu_uninit_channel(MEM_BG_SYNC);
+#endif
+
 	/*
 	 * this function is called just before we call linux
 	 * it prepares the processor for linux
-- 
1.5.4.4

